### PR TITLE
remove reuse logits for grads from python interface.

### PR DESCRIPTION
**Pull request by vincentqb**
Follow up to https://github.com/pytorch/audio/pull/1532 (and needs to lands after #1532)

The default for `reuse_logits_for_grads=True` can also be confusing to users and lead to the inputs being modified in place.
- This is not the usual behavior of a loss function and needs to be changed. 
- When a function changes its inputs in place, it also needs to report this to autograd by marking them as dirty.

I’ll simply remove this option for now.

### CODE REVIEW

**Comment by carolineechen** [comment]
we should also make sure to remove this parameter from the tests and c++ source code as well

**Comment by vincentqb** [comment]
Closed by [comment](https://github.com/pytorch/audio/pull/1532#discussion_r647660456).
